src/transaction.ts

```diff
@@ -62,22 +62,30 @@ export function assembleTransaction(
   switch (raw.operations[0].type) {
     case "invokeHostFunction":
       const invokeOp: Operation.InvokeHostFunction = raw.operations[0];
+      const existingAuth = invokeOp.auth ?? [];
       txnBuilder.addOperation(
         Operation.invokeHostFunction({
           source: invokeOp.source,
           func: invokeOp.func,
-          // apply the auth from the simulation
-          auth: (invokeOp.auth ?? []).concat(
-            simulation.results[0].auth?.map((a: string) =>
-              xdr.SorobanAuthorizationEntry.fromXDR(a, "base64")
-            ) ?? []
-          ),
+          // if auth entries are already present, we consider this "advanced
```

Comment:
minor suggestion, this is really good description on the new behavior for auth based on incoming tx, which could be included in public facing js doc on [prepareTransaction()](https://github.com/stellar/js-soroban-client/pull/114/files#diff-8a8ae07582c9d433ec8c2e5c4310ff8901e604f4965c5b90a49117ad46c47595R466)

Reply:
great idea! done in 12973ee